src/pyhf/cli/patchset.py

```diff
@@ -16,6 +21,42 @@ def cli():
     """Operations involving patchsets."""
 
 
+@cli.command()
+@click.argument("archive-url", default="-")
+@click.argument("output-directory", default="-")
+@click.option("-v", "--verbose", is_flag=True, help="Enables verbose mode")
+@click.option(
+    "-f", "--force", is_flag=True, help="Force download from non-approved host"
+)
+def download(archive_url, output_directory, verbose, force):
+    """
+    Download the patchset archive from the remote URL and extract it in a directory at the path given.
+
+    Raises:
+        :class:`~pyhf.exceptions.InvalidArchiveHost`: if the provided archive host name is not known to be valid
+
+    Returns:
+        None
+    """
+    if not force:
+        hostname = archive_url.split("://")[1].split("/")[0].strip("www.")
+        valid_hosts = ["hepdata.net", "doi.org"]
+        if hostname not in valid_hosts:
+            raise exceptions.InvalidArchiveHost(
+                f"{hostname} is not an approved archive host: {', '.join(str(host) for host in valid_hosts)}\n"
+                + "To download an archive from this host use the --force option."
+            )
+
+    req = urllib.request.Request(archive_url)
+    with urllib.request.urlopen(req) as response:
```

Comment:
@kratsg @lukasheinrich I don't understand why Bandit is complaining here with the 
> Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected. (B310)
below

The [Bandit blacklist page doesn't really make things more clear either](https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b310-urllib-urlopen). Does this make sense to you?

Reply:
it means you need to make sure you trust the url you're opening.

Reply:
Sure. Is there a way to more properly validate this then what I'm doing now?

Reply:
https://requests.readthedocs.io/en/master/ essentially... It's the only one not ban-listed in bandit's list. Or we add a `.bandit` config file.